builtin/providers/vcd/resource_vcd_edgegateway_vpn.go

```diff
@@ -0,0 +1,281 @@
+		PeerSubnet:         peerSubnets,
+		SharedSecret:       d.Get("shared_secret").(string),
+		IsEnabled:          true,
+	}
+
+	tunnels := make([]*types.GatewayIpsecVpnTunnel, 1)
+	tunnels[0] = tunnel
+
+	ipsecVPNConfig := &types.EdgeGatewayServiceConfiguration{
+		Xmlns: "http://www.vmware.com/vcloud/v1.5",
+		GatewayIpsecVpnService: &types.GatewayIpsecVpnService{
+			IsEnabled: true,
+			/*
+				Endpoint: &types.GatewayIpsecVpnEndpoint{
+					Network: &types.Reference{
+						HREF: "http://myvpn.com",
+					},
+					PublicIP: "63.30.253.57",
+				},
+			*/
+			Tunnel: tunnels,
+		},
+	}
+
+	log.Printf("[INFO] ipsecVPNConfig: %#v", ipsecVPNConfig)
+
+	err = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
+		edgeGateway.Refresh()
+		task, err := edgeGateway.AddIpsecVPN(ipsecVPNConfig)
+		if err != nil {
+			log.Printf("[INFO] Error setting ipsecVPNConfig rules: %s", err)
+			return resource.RetryableError(
+				fmt.Errorf("Error setting ipsecVPNConfig rules: %#v", err))
+		}
+
+		return resource.RetryableError(task.WaitTaskCompletion())
+	})
+	if err != nil {
+		return fmt.Errorf("Error completing tasks: %#v", err)
+	}
+
+	d.SetId(d.Get("edge_gateway").(string))
+
+	return resourceVcdEdgeGatewayVpnRead(d, meta)
+}
+
+func resourceVcdEdgeGatewayVpnDelete(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+
+	log.Printf("[TRACE] CLIENT: %#v", vcdClient)
+
+	vcdClient.Mutex.Lock()
+	defer vcdClient.Mutex.Unlock()
+
+	edgeGateway, err := vcdClient.OrgVdc.FindEdgeGateway(d.Get("edge_gateway").(string))
+
+	ipsecVPNConfig := &types.EdgeGatewayServiceConfiguration{
+		Xmlns: "http://www.vmware.com/vcloud/v1.5",
+		GatewayIpsecVpnService: &types.GatewayIpsecVpnService{
+			IsEnabled: false,
+		},
+	}
+
+	log.Printf("[INFO] ipsecVPNConfig: %#v", ipsecVPNConfig)
+
+	err = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
+		edgeGateway.Refresh()
+		task, err := edgeGateway.AddIpsecVPN(ipsecVPNConfig)
+		if err != nil {
+			log.Printf("[INFO] Error setting ipsecVPNConfig rules: %s", err)
+			return resource.RetryableError(
+				fmt.Errorf("Error setting ipsecVPNConfig rules: %#v", err))
+		}
+
+		return resource.RetryableError(task.WaitTaskCompletion())
+	})
+	if err != nil {
+		return fmt.Errorf("Error completing tasks: %#v", err)
+	}
+
+	d.SetId(d.Get("edge_gateway").(string))
+
+	if err != nil {
+		return fmt.Errorf("Error finding edge gateway: %#v", err)
+	}
+
+	return nil
+}
+
+func resourceVcdEdgeGatewayVpnRead(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+
+	edgeGateway, err := vcdClient.OrgVdc.FindEdgeGateway(d.Get("edge_gateway").(string))
+	if err != nil {
+		return fmt.Errorf("Error finding edge gateway: %#v", err)
+	}
+
+	fmt.Println(edgeGateway)
+
+	//d.Set("name", *edgeGateway.EdgeGateway.Configuration.EdgeGatewayServiceConfiguration.GatewayIpsecVpnService.Tunnel[0].Name)
+	// and all the others
```

Comment:
The attributes inside of the schema will need to be populated with the result returned from the API here